src/App.tsx

```diff
@@ -1,25 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { getPageNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+  const numberPages: number[] = getPageNumbers(items.length, perPage);
+  const currentIndex = numberPages.indexOf(currentPage);
+  const fromIndex = currentIndex * perPage;
+  const toIndex =
+    fromIndex + perPage <= items.length ? fromIndex + perPage : items.length;
+
+  const handlePerPage = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
+  };
```

Comment:
Congratulations!

Nice logic you got!
A next step here is copy all these code and make a custom hook `usePagination` that exports the used constants. (you don't need to do it now, just study the concept)

Reply:
Thanks! 
Custom hook is new to me! I'll take a look how it works!! Thank you!